transform/utils.go

```diff
@@ -15,7 +15,7 @@ import (
 var indexRe = regexp.MustCompile(`\[([\d]+)\]`)
 
 // Concat will combine any two arbitrary values, though only strings are supported for non-trivial concatenation.
-func concat(a, b interface{}) (interface{}, error) {
+func concat(a, b, delimiter interface{}) (interface{}, error) {
```

Comment:
I think that the delimiter should be a string. I think the reason that a and b are empty interface is that there type is not known but that isn't true of the delimiter.